js/main.js

```diff
@@ -0,0 +1,214 @@
+
+  var fatsP = document.createElement('p');
+  fatsP.textContent = 'Fats: ' + data.dailyTotals.fats + '/' + data.targets.fats + ' g';
+  fatsColumnThird.appendChild(fatsP);
+
+  var fatsTwoThirds = document.createElement('div');
+  fatsTwoThirds.setAttribute('class', 'column-66 padding-right');
+  fatsRow.appendChild(fatsTwoThirds);
+
+  var fatsProgressBarDiv = document.createElement('div');
+  fatsProgressBarDiv.setAttribute('class', 'progress-bar background-color-white row align-items-center');
+  fatsTwoThirds.appendChild(fatsProgressBarDiv);
+
+  var fatsProgressFillDiv = document.createElement('div');
+  fatsProgressFillDiv.setAttribute('style', 'width: ' + data.dailyTotals.fats / data.targets.fats * 100 + '%');
+  fatsProgressFillDiv.setAttribute('class', 'fill-progress-fats progress text-align-center padding-top-3');
+  fatsProgressBarDiv.appendChild(fatsProgressFillDiv);
+
+  var fatsProgressFillText = document.createElement('p');
+  fatsProgressFillText.setAttribute('class', 'margin-top-0 color-white font-weight-bold');
+  fatsProgressFillText.textContent = fatsProgressFillDiv.style.width;
+  fatsProgressFillDiv.appendChild(fatsProgressFillText);
+
+  // Carbs
+
+  var carbohydratesRow = document.createElement('div');
+  carbohydratesRow.setAttribute('class', 'row align-items-center');
+  dataViewDiv.appendChild(carbohydratesRow);
+
+  var carbohydratesColumnThird = document.createElement('div');
+  carbohydratesColumnThird.setAttribute('class', 'column-33');
+  carbohydratesRow.appendChild(carbohydratesColumnThird);
+
+  var carbohydratesP = document.createElement('p');
+  carbohydratesP.textContent = 'Carbs: ' + data.dailyTotals.carbohydrates + '/' + data.targets.carbohydrates + ' g';
+  carbohydratesColumnThird.appendChild(carbohydratesP);
+
+  var carbohydratesTwoThirds = document.createElement('div');
+  carbohydratesTwoThirds.setAttribute('class', 'column-66 padding-right');
+  carbohydratesRow.appendChild(carbohydratesTwoThirds);
+
+  var carbohydratesProgressBarDiv = document.createElement('div');
+  carbohydratesProgressBarDiv.setAttribute('class', 'progress-bar background-color-white row align-items-center');
+  carbohydratesTwoThirds.appendChild(carbohydratesProgressBarDiv);
+
+  var carbohydratesProgressFillDiv = document.createElement('div');
+  carbohydratesProgressFillDiv.setAttribute('style', 'width: ' + (data.dailyTotals.carbohydrates / data.targets.carbohydrates * 100) + '%');
+  carbohydratesProgressFillDiv.setAttribute('class', 'fill-progress-carbohydrates progress text-align-center padding-top-3');
+  carbohydratesProgressBarDiv.appendChild(carbohydratesProgressFillDiv);
+
+  var carbohydratesProgressFillText = document.createElement('p');
+  carbohydratesProgressFillText.setAttribute('class', 'margin-top-0 color-white font-weight-bold');
+  carbohydratesProgressFillText.textContent = carbohydratesProgressFillDiv.style.width;
+  carbohydratesProgressFillDiv.appendChild(carbohydratesProgressFillText);
+
```

Comment:
This still isn't returning the DOM tree it generated, you should be returning the DOM tree and appending it where the function is called.